src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -425,13 +425,18 @@ private Territory getRetreatInternal(final String message, final Collection<Terr
       // if you have eliminated the impossible, whatever remains, no matter
       // how improbable, must be the truth
       // retreat
-      final RetreatComponent comp = new RetreatComponent(possible);
-      final int option = JOptionPane.showConfirmDialog(BattleDisplay.this, comp, message,
-          JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE, null);
-      if (option == JOptionPane.OK_OPTION) {
-        if (comp.getSelection() != null) {
-          retreatTo[0] = comp.getSelection();
-          latch.countDown();
+      if (possible.size() == 1) {
+        retreatTo[0] = possible.iterator().next();
+        latch.countDown();
```

Comment:
You're correct. My example was overly complicated and not sure why I was thinking of another battle and not just multiple owned adjacent territories.